service/src/main/java/com/wso2telco/dep/tpservice/pool/TokenReGenarator.java

```diff
@@ -98,21 +112,55 @@ public TokenDTO reGenarate(final WhoDTO who, final TokenDTO oldToken) throws Tok
 		// validate response message
 		if (Strtoken != null && Strtoken.length() > 0) {
 
-			JSONObject jsontoken = new JSONObject(Strtoken);
+			try {
+				jsontoken = new JSONObject(Strtoken);
```

Comment:
Then check the responseCode at that scenario and include them as mandatory response codes because even if u check 502 or 503 scenarios, html response is returned.